### ISSUE TITLE
Can menu bar set in Caption\Title Bar?

### ISSUE THREAD

**Issue by jim-jiang-github** [dotnet/maui#7160]
### Description

Is there any way to set the menu bar to the caption on windows?
I try set the menu bar in 'ContentPage' and got the result:
![image](https://user-images.githubusercontent.com/29189410/168296884-14b72b85-1d64-43a2-997f-ea75c86d44d4.png)

And another way is set the menu bar to 'Platforms' folder's App.xaml, And I got this:
![image](https://user-images.githubusercontent.com/29189410/168297710-4f122e99-bff4-4f67-9d0c-409dbad2bded.png)

It looks beautiful, But how can I Trigger the menu event or binding the command???

### Public API Changes

```
 <ContentPage.MenuBarItems>
        <MenuBarItem Text="File">
            <MenuFlyoutItem Text="Open" />
            <MenuFlyoutItem Text="Save" />
            <MenuFlyoutItem Text="Exit" />
        </MenuBarItem>
        <MenuBarItem Text="Edit">
            <MenuFlyoutSubItem Text="Rotate">
                <MenuFlyoutItem Text="Rotate 90" />
                <MenuFlyoutItem Text="Rotate 180" />
                <MenuFlyoutItem Text="Rotate 270" />
                <MenuFlyoutItem Text="Rotate 360" />
            </MenuFlyoutSubItem>
            <MenuFlyoutItem Text="Zoom" />
            <MenuFlyoutItem Text="Reset" />
        </MenuBarItem>
    </ContentPage.MenuBarItems>
```

### Intended Use-Case

Can menu bar set in caption?

**Comment by PureWeen** [comment]
@jim-jiang-github that's the eventual plan :-) We're waiting on some additional behavior to make its way to WinAppSDK and then we'll enable the option of moving the menu up into the title bar

**Comment by Jarek300** [comment]
Does it mean that WinUI 3 title bar customization described [here](https://docs.microsoft.com/en-us/windows/apps/develop/title-bar?tabs=wasdk) will be supported by Maui Windows application? In .NET 7?

**Comment by mattleibow** [comment]
@PureWeen is htis maybe something that we can do with the "extends into titlebar" code as it looks like the issue here is just putting the menu in the title? Or is this different? 

Also @jim-jiang-github what is your idea about dragging and double clicking to maximize if the menu is in the title bar? Will the menu still work and maybe you are thinking for the menu is a menu, but the remaining space is for the user interaction?

@PureWeen do we have a MAUI issue tracking this new feature of total titlebar customization? Is this a missing feature in the WinUI SDK or just not yet in MAUI?

**Comment by mattleibow** [comment]
Could this be linked to the other issue #7123

**Comment by jfversluis** [comment]
I think with the new [TitleBar](https://learn.microsoft.com/dotnet/maui/user-interface/controls/titlebar?view=net-maui-9.0) control you should be able to do a lot of what you want. If you want to see anything specific, please open a new issue focussing on that scenario. Thanks!
